[ALS-4284] Added documentation to assist a user in configuring their …

Gcolon021 opened the pull request:
Added a guide that can be followed by a user in order to setup their front-end harness

Gcolon021 committed:
[ALS-4284] Added documentation to assist a user in configuring their frontend harness environment on a macOS.

srpiatt commented on README.md:
```diff
@@ -7,14 +7,62 @@ each change. These instructions assume you are on a Mac and that you have Docker
-Issues that sometimes happen:
+5. Move the `settings.json` file that was downloaded previously into the `$(pwd)/pic-sure-frontend-harness/repos/base_settings/` directory. <b>Note</b>: If the directory `base_settings` does not exist create it.
+
+6. Navigate to pic-sure-frontend-harness/repos and clone the baseline-pic-sure repository by running git clone https://github.com/hms-dbmi/baseline-pic-sure.
```
```suggestion
6. Navigate to `pic-sure-frontend-harness/repos` and clone the baseline-pic-sure repository by running `git clone https://github.com/hms-dbmi/baseline-pic-sure`.
```
Matching your formatting, above.

srpiatt reviewed (approved):
Looks great! (Merge after reviewing in office-hours, like normal. I'm just approving to show my approval lol)

Gcolon021 committed:
Update README.md

Co-authored-by: Samantha <samantha.r.piatt@gmail.com>

JamesPeck commented on README.md:
```diff
@@ -7,14 +7,62 @@ each change. These instructions assume you are on a Mac and that you have Docker
-Steps to configure:
+### Preface
+Before starting, ensure that you have set up your PIC-SURE All-In-One virtual box environment with a bridge network, 
+that you can navigate to your PIC-SURE environment using the IP Address on your virtual box and have logged in. 
+Additionally, note that this guide was created for the macOS.
 
-./test_using_remote_backend.sh
+### Guide
+1. Visit your virtualbox PIC-SURE All-In-One page, inspect, and open the network tab.
+    1. Reload the page to allow the network tab to populate.
+    2. Search for and download `settings.json`.
+2. Open the browser certificates for the current page and download `picsure.local.cert`.
+    1. Add `picsure.local.cert` to your local security chain.
+        1. Double-click on the `picsure.local` in your security chain.
+        2. Expand the `Trust` section.
+        3. Select  `When using this certificate: Always Trust`.
```
The frontend harness isn't really related to the all in one. The frontend harness's  role is really for developing using a remote backend, for example some ec2 running one of our httpd servers.This section should be removed. 
I just had you set yours up to point to your local all in one because then you control everything.

JamesPeck replied:
We can make a guide for internal use to do these steps

Gcolon021 replied:
I will separate it out into a google doc for internal use.

JamesPeck commented on README.md:
```diff
@@ -7,14 +7,62 @@ each change. These instructions assume you are on a Mac and that you have Docker
+
+10. Navigate to the `cert` directory.
+     1. Add `localhost.ca.pem` to your local security chain.
+         1. Double-click on the `localhost.ca`. In your security chain.
+         2. Expand the `Trust` section.
+         3. Select  `When using this certificate: Always Trust`
```
in md files you can just use 

 1. example

and then

1. example

and it automatically does 1,2,3 etc. This way its easier to add/remove steps in the future

JamesPeck commented on README.md:
````diff
@@ -7,14 +7,62 @@ each change. These instructions assume you are on a Mac and that you have Docker
+
+7. Open `test_using_remote_backend.sh` with a text editor and add the following lines to the top of the file, but just below `#!/bin/bash`:
+    ```
+    export BACKEND_HOST=picsure.local
+    export BACKEND_IP=<YOUR VIRTUALBOX IP ADDRESS>
+    export PROJECT_SPECIFIC_UI_PATH=repos/baseline-pic-sure/ui  
+    
+    export ADDITIONAL_VOLUMES=" -v $(pwd)/repos/base_settings/settings.json:/usr/local/apache2/htdocs/picsureui/settings/settings.json "
````
We should make this more generic

JamesPeck commented on README.md:
````diff
@@ -7,14 +7,62 @@ each change. These instructions assume you are on a Mac and that you have Docker
+8. Open your host file: `vim /private/etc/hosts`
+    1. Modify the file to remove or comment out your previous
+       `<virtualbox IP Address> picsure.local`
+       1. Add a new record to reflect the following: `127.0.0.1 picsure.local`. The file should look similar to the following lines:
+           ```
+           # PIC-SURE All-In-One Virtualbox IP Address
+           #192.168.1.63   picsure.local
+        
+           # PIC-SURE UI Harness
+           127.0.0.1       picsure.local
+           ```
````
Probably should be generic again

Gcolon021 committed:
[ALS-4284] Removed sections that referenced the All-In-One Virtual box as it is to specific, and it is not a required when using the harness. It The harness can be utilized with any remote backend. Updated the section that covers modifying the test_using_remote_back.sh script to be more general.
Modified the numbering approach to use all "1." letting md handle the numbering making for a more maintain document.

Gcolon021 committed:
[ALS-4284] Updated steps 2, 3, and 4 to be more generally applicable. Updated step 6 to remove a ß.

Gcolon021 committed:
Update README.md

Found another ß

JamesPeck commented on README.md:
````diff
@@ -38,17 +43,18 @@ Before starting, note that this guide was created for the macOS.
+    # Example: export ADDITIONAL_VOLUMES=" -v $(pwd)/repos/studies-data.json:/usr/local/apache2/htdocs/picsureui/studyAccess/studies-data.json "
+    #          export ADDITIONAL_VOLUMES=" -v $(pwd)/repos/path_a:/usr/local/apache2/htdocs/path_a -v $(pwd)/path_b:/usr/local/apache2/htdocs/path_b "
+    export ADDITIONAL_VOLUMES=" -v $(pwd)/repos/settings.json:/usr/local/apache2/htdocs/picsureui/settings/settings.json "
+   ```ß
````
hahaha

Gcolon021 replied:
I just found it too!
